addOns/exim/src/main/java/org/zaproxy/addon/exim/har/MenuImportHar.java

```diff
@@ -63,19 +65,23 @@ public void run() {
                                         File file = chooser.getSelectedFile();
                                         int tasks = 0;
                                         boolean indeterminate;
+                                        HarLog log = null;
                                         try {
-                                            tasks =
-                                                    new HarFileReader()
-                                                            .readHarFile(file)
-                                                            .getEntries()
-                                                            .getEntries()
-                                                            .size();
+                                            log = new HarReader().readFromFile(file).getLog();
+                                            tasks = log.getEntries().size();
                                             indeterminate = false;
-                                        } catch (IOException e) {
+                                        } catch (HarReaderException e) {
                                             indeterminate = true;
                                             LOGGER.warn(
-                                                    "Couldn't count entries in: {}",
-                                                    file.getAbsoluteFile());
+                                                    "Failed to read HAR file: {} \n {}",
+                                                    file.getAbsolutePath(),
+                                                    e.getMessage());
+                                            Stats.incCounter(
+                                                    ExtensionExim.STATS_PREFIX
+                                                            + HarImporter.STATS_HAR_FILE_ERROR);
+                                        }
+                                        if (log == null) {
+                                            return;
```

Comment:
This changes the existing logic, it can no longer be indeterminate.

Reply:
Because it's always a list now so we always have a known number.

Reply:
Right and why the code should be changed as it can no longer be indeterminate.

Reply:
This was not changed.

Reply:
Oh crud, I think I removed the exact same condition in another spot because I didn't realize this comment was on a different class 🤦‍♂️

Reply:
I noticed and I was going to ask if that null check was no longer needed, but didn't associate that change with this review comment.

Reply:
Restored the incorrectly removed one, still need to address the comment for the Menu class.

Reply:
I adjusted the code here. 'Should' be good now 😉